feature/foryou/src/main/java/com/google/samples/apps/nowinandroid/feature/foryou/ForYouScreen.kt

```diff
@@ -135,119 +116,88 @@ internal fun ForYouScreen(
-                interestsSelectionState = interestsSelectionState,
-                onAuthorCheckedChanged = onAuthorCheckedChanged,
-                onTopicCheckedChanged = onTopicCheckedChanged,
-                saveFollowedTopics = saveFollowedTopics,
-                // Custom LayoutModifier to remove the enforced parent 16.dp contentPadding
-                // from the LazyVerticalGrid and enable edge-to-edge scrolling for this section
-                interestsItemModifier = Modifier.layout { measurable, constraints ->
-                    val placeable = measurable.measure(
-                        constraints.copy(
-                            maxWidth = constraints.maxWidth + 32.dp.roundToPx()
-                        )
+    LazyVerticalGrid(
+        columns = Adaptive(300.dp),
+        contentPadding = PaddingValues(16.dp),
+        horizontalArrangement = Arrangement.spacedBy(16.dp),
+        verticalArrangement = Arrangement.spacedBy(24.dp),
+        modifier = modifier
+            .fillMaxSize()
+            .testTag("forYou:feed"),
+        state = state
+    ) {
+        interestsSelection(
+            interestsSelectionState = interestsSelectionState,
+            onAuthorCheckedChanged = onAuthorCheckedChanged,
+            onTopicCheckedChanged = onTopicCheckedChanged,
+            saveFollowedTopics = saveFollowedTopics,
+            // Custom LayoutModifier to remove the enforced parent 16.dp contentPadding
+            // from the LazyVerticalGrid and enable edge-to-edge scrolling for this section
+            interestsItemModifier = Modifier.layout { measurable, constraints ->
+                val placeable = measurable.measure(
+                    constraints.copy(
+                        maxWidth = constraints.maxWidth + 32.dp.roundToPx()
                     )
-                    layout(placeable.width, placeable.height) {
-                        placeable.place(0, 0)
-                    }
+                )
+                layout(placeable.width, placeable.height) {
+                    placeable.place(0, 0)
                 }
-            )
+            }
+        )
 
-            newsFeed(
-                feedState = feedState,
-                onNewsResourcesCheckedChanged = onNewsResourcesCheckedChanged,
-            )
+        newsFeed(
+            feedState = feedState,
+            onNewsResourcesCheckedChanged = onNewsResourcesCheckedChanged,
+        )
 
-            item(span = { GridItemSpan(maxLineSpan) }) {
+            /*item(span = { GridItemSpan(maxLineSpan) }) {
                 Column {
                     Spacer(modifier = Modifier.height(8.dp))
                     // Add space for the content to clear the "offline" snackbar.
+                    // TODO: Check that the Scaffold handles this correctly in NiaApp
                     if (isOffline) Spacer(modifier = Modifier.height(48.dp))
                     Spacer(Modifier.windowInsetsBottomHeight(WindowInsets.safeDrawing))
```

Comment:
Removing the `windowInsetsBottomHeight` here means that the cards might overlap with the navigation bar insets when the navigation rail is used. Can we bring that back (even if we leave the TODO for the spacer that accounts for the snackbar?)

Reply:
Thanks for spotting this. My mistake. I've added everything except the spacer for the snackbar back in.